tests/suites/test_suite_pk.function

```diff
@@ -208,8 +206,24 @@ static int pk_setup_for_type(mbedtls_pk_type_t pk_type, int want_pair,
             *psa_type = PSA_KEY_TYPE_ECC_KEY_PAIR(mbedtls_ecc_group_to_psa(grp_id, &bits));
             TEST_EQUAL(pk_genkey(pk, grp_id), 0);
             if (!want_pair) {
+#if defined(MBEDTLS_PK_USE_PSA_EC_DATA)
+                psa_key_attributes_t pub_attributes = PSA_KEY_ATTRIBUTES_INIT;
+                psa_set_key_type(&pub_attributes,
+                                 PSA_KEY_TYPE_PUBLIC_KEY_OF_KEY_PAIR(*psa_type));
+                psa_set_key_usage_flags(&pub_attributes,
+                                        PSA_KEY_USAGE_EXPORT |
+                                        PSA_KEY_USAGE_COPY |
+                                        PSA_KEY_USAGE_VERIFY_MESSAGE |
+                                        PSA_KEY_USAGE_VERIFY_HASH);
+                psa_set_key_algorithm(&pub_attributes, PSA_ALG_ECDSA_ANY);
+                PSA_ASSERT(psa_destroy_key(pk->priv_id));
+                PSA_ASSERT(psa_import_key(&pub_attributes,
+                                          pk->pub_raw, pk->pub_raw_len,
+                                          &pk->priv_id));
```

Comment:
That's not how `USE_PSA_EC_DATA` usually store public keys. As the name suggests, `priv_id` is only used for private keys. (The reason is we don't want to use a PSA key slot for each public key because X.509 may create a lot of `pk_context`s for public keys, see "The problem of X.509 vs the PSA key store" in #8452.) See the comments in `pk.h`.

See also how keys are created in `pkparse.c`, specifically `pk_ecc_set_pubkey()` vs `pk_ecc_set_key()`. I think the keys created here should look the same as those crated by pkparse - because that's how most `pk_context`s are populated in practice.

Actually this provides a perfect transition to another thought that's been on my mind lately: is generating keys for each test case really a reasonable strategy? Shouldn't we just parse instead? Would be better for performance (key generation is computationally intensive) and reduce the amount of test code to maintain. (I'll note for RSA public keys in a previous commit you used `import()` not `generate()`. The downside is, for ECC there are a lot of curves, so do we want to restrict ourselves to just one per family, or embed test data for each curve?

Probably out of scope for this PR, but I still wanted to mention it.

Reply:
> is generating keys for each test case really a reasonable strategy? Shouldn't we just parse instead?

I think parsing would be better. I didn't implement it because I wanted to reuse the existing code and I didn't want to add a depdendency on pkparse. With https://github.com/Mbed-TLS/mbedtls/pull/8740 we'll be in a much better position. We may want to clean up how test data is constructed here before we do further work in pk. Cc @valeriosetti

Reply:
Note: we could use parsing for RSA and generation for ECC. Generation has the advantage that it can adapt to any set of curves, and for ECC key pairs it's not a performance hit (and even for ECC public keys, the performance hit it pretty minor).

Reply:
How can I create a pk object with an EC public key? The only way to set the EC-specific fields seems to be `mbedtls_pk_setup_opaque`, and it only works on private keys.

(I've handled or punted on all your feedback except this, which confuses me.)

Reply:
In the `USE_PSA_EC_DATA` case, to create a PK context holding an EC public key, you just set `pub_raw` and `pub_raw_len` to the canonical PSA representation of the public key (ie, something that can be passed to `psa_import_key()` when the time comes to actually use the key) and `ec_family` and `ec_bits` to represent the curve.

Perhaps your question means [the comments in `pk.h`](https://github.com/Mbed-TLS/mbedtls/blob/development/include/mbedtls/pk.h#L230) are not as clear as they should be. If find them OK but I'm obviously not a good judge here, being too familiar with the topic myself. Feel free to suggest improvements or point out where you think documentation is lacking.

Reply:
I found the comments in `pk.h` clear as to how to interpret the fields, but not as to exactly what combinations are possible. What's `pk_info->` in that case? What's missing from the test data to reveal this omission in the library?